awscli/customizations/s3/subcommands.py

```diff
@@ -1359,6 +1382,19 @@ def _raise_if_mv_same_paths(self, src, dest):
                 f"{self.parameters['src']} - {self.parameters['dest']}"
             )
 
+    def _raise_if_paths_type_incorrect_for_param(self, param, paths_type, expected_paths_type):
+        if paths_type != expected_paths_type:
```

Comment:
Instead of `expected_paths_type` let's make the arg a list and call it  `allowed_paths_types`. And here we can make the check `if paths_type not in allowed_paths_types`.